framework/util/spirv_parsing_util.cpp

```diff
@@ -0,0 +1,476 @@
+                                                                                           : "unknown");
+                                }
+                                else
+                                {
+                                    GFXRECON_LOG_WARNING("Access-chain index is out-of-bounds for op: %s",
+                                                         string_SpvOpcode(td->op));
+                                    return;
+                                }
+                            }
+                            access_indices.clear();
+
+                            if (td->op == SpvOpTypeRuntimeArray)
+                            {
+                                buffer_reference_info.array_stride = td->traits.array.stride;
+                            }
+
+                            // buffer-references traced back to either pointer-type, uin64_t or arrays of those
+                            if (td->op == SpvOpTypeForwardPointer ||
+                                (td->op == SpvOpTypeInt && td->traits.numeric.scalar.width == 64) ||
+                                td->op == SpvOpTypeRuntimeArray)
+                            {
+                                buffer_reference_map_[buffer_reference_info] = access_chain_names;
+                            }
+                            else
+                            {
+                                GFXRECON_LOG_WARNING(
+                                    "Traced back a potential buffer-reference, but type does not match: %s",
+                                    string_SpvOpcode(td->op));
+                            }
+                        }
+                        object_insn = nullptr;
+                    }
+                    break;
+                }
+                default:
+                    GFXRECON_LOG_WARNING("Failed to track back the Function Variable OpStore, hit a %s\n",
+                                         string_SpvOpcode(object_insn->opcode()));
+                    object_insn = nullptr;
+                    break;
+            }
+        }
+    };
+
+    // Now we can walk the SPIR-V one more time to find what we need
+    for (const Instruction& insn : instructions)
+    {
+        // because it is SSA, we can build this up as we are looping in this pass
+        const uint32_t result_id = insn.resultId();
+        if (result_id != 0)
+        {
+            definitions_[result_id] = &insn;
+        }
+
+        const uint32_t opcode = insn.opcode();
+
+        if (opcode == spv::OpStore)
+        {
+            store_instructions_.push_back(&insn);
+        }
+        else if (opcode == spv::OpDecorate)
+        {
+            decorations_instructions_.push_back(&insn);
+        }
+
+        // There is always a load that does the dereferencing
+        if (opcode != spv::OpLoad)
+        {
+            continue;
```

Comment:
just noting that the `Parse` function currently is heavily aimed toward buffer device addressing, which I think is totally fine, but something to caution if someone wanted to add something different to this as well

Reply:
yes, good point. maybe just name it `ParseBufferReferences` since that all it currently does

Reply:
ya, mainly to give the future person a fair idea it is not just a simple "add my small code in it"